# Practica Calificada N°1

**Problema N°1**

Una muestra de 90 clientes de cierta gasolinera indica que el número medio de galones comprados es de 15.2 y la desviación estándar de 2.7 galones. Construya un intervalo de confianza del 95 % para el número medio de galones de gasolina comprados.

**Se empleara:**
$$
IC{:}\quad\bar{X}-Z_{(1-\frac\alpha2)}\cdot\frac\sigma{\sqrt{n}}\leq\mu\leq\bar{X}+Z_{(1-\frac\alpha2)}\cdot\frac\sigma{\sqrt{n}}
$$

Los parámetros relevantes son:
- Varianza Poblacional ( $\sigma$ ) = 2.7
- Promedio ($\bar{X}$) = 15.2
- Tamaño de muestra ($n$) = 90
- Confianza = 95 %

**Si:**
1 - $\alpha$ = 0.95

**Entonces:**
- $\alpha$ = 0.05

In [2]:
import scipy.stats as st
import math
# Asignando valores:
O = 2.7
X = 15.2
n = 90
# a -> calculado previamente
a = 0.05
Za= 1-(a/2)

# Calculamos Z
Z = round((st.norm.ppf(Za)),2)

# Reemplazamos en la formula IC
IC1 = X-(Z*O/math.sqrt(n)) 
IC2 = X+(Z*O/math.sqrt(n)) 

print(Za)
print(Z)
print("IC:",IC1," u ",IC2)

0.975
1.96
IC: 14.642174220746297  u  15.757825779253702


Con un nivel de confianza del 95%, el numero medio de galones de gasolina comprados está comprendido desde 14.642 a 15.757.

---

**Ejercicio N°2**

Se quiere comparar el gasto mensual en arbitrios de dos ciudades A y B, por lo cual se toma una muestra al azar de 200 hogares de cada ciudad, donde la media de la ciudad A resulto S/. 350 y la de B; S/. 230. Si se sabe que la desviación poblacional en A es de S/. 40 y en B de S/. 30, calcule el intervalo de confianza a un 95% para la diferencia de medias e indique si el gasto medio en ambas ciudades es el mismo. 

Los parametros relevantes son:
- Varianza Poblacional (Desviacion) ($\sigma_1^2$) = 40
- Muestra 1 ($n_1$) = 200 
- Promedio 1 ($\bar{X}_1$) = 350
- Varianza Poblacional (Desviacion) ($\sigma_2^2$) = 30
- Muestra 2 ($n_2$) = 200
- Promedio 2 ($\bar{X}_2$) = 230
- Confianza = 95%

**Formula a utilizar:**

$$
IC=(X_1-X_2)-Z_{(1-\frac{\alpha}{2})}\sqrt[]{\frac{\sigma_1{}^2}{n_1}+\frac{\sigma_2{}^2}{n_2}}\leq\boldsymbol{\mu_1}-\boldsymbol{\mu_2}\leq(X_1-X_2)+Z_{(1-\frac{\alpha}{2})}\sqrt[]{\frac{\sigma_1{}^2}{n_1}+\frac{\sigma_2{}^2}{n_2}}
$$

**Si:**
1 - $\alpha$ = 0.95

**Entonces:**
- $\alpha$ = 0.05

In [6]:
import scipy.stats as st
import math

# Asignando valores:
o1 = 40
o2 = 30
n1 = 200
n2 = 200
x1 = 350
x2 = 230
# a -> calculado previamente
a = 0.05
Za=1 - a / 2

# Calculamos Z
Z = round(st.norm.ppf(Za), 3)

# Reemplazamos en la fórmula IC
IC1 = round((x1 - x2) - Z * (math.sqrt(o1**2 / n1 + o2**2 / n2)), 3)
IC2 = round((x1 - x2) + Z * (math.sqrt(o1**2 / n1 + o2**2 / n2)), 3)

print("Za:",Za)
print("Z:", Z)
print("Intervalo de confianza:", IC1, "<= \u03BC <=", IC2)

Za: 0.975
Z: 1.96
Intervalo de confianza: 113.07 <= μ <= 126.93


Con un nivel de confianza del 95% podemos decir que la diferencia de medias de los gastos en arbitrios de las ciudades A y B se encuentran en el intervalo (113.07 y 126.93)

---

**Problema N°3**

Un proceso está programado para embolsar la cantidad media gramos de café. Últimamente se ha visto que la maquina está embolsando menos, por tal razón se toma una muestra aleatoria de 100 bolsas, resultando que 26 de ellas tienen menos de 250 gramos. Construya un intervalo de confianza del 95% para la proporción de bolsas con menor peso de lo deseado. 

Parametros resaltantes:
- Muestra ($n$) = 100
- Numero de exitos en la muestra ($X$) = 26
- Confianza = 95%

**Formulas a utilizar:**
$$
IC=p-Z_{\left(1-\frac\alpha2\right)^\bullet}\sqrt{\frac{pq}n}<{\pi}<p+Z_{\left(1-\frac\alpha2\right)^\bullet}\sqrt{\frac{pq}n}
$$

A favor:
$$
p=\frac{X}{n}
$$
En contra:
$$
q = 1 - p 
$$

In [11]:
import scipy.stats as st
import math

# Asignando valores:
n=100
X=26
CF=0.95
p=X/n
q=1-p
a = round(1 - CF,3)
Za=1-(a/2)

# Calculamos Z
Z = round(st.norm.ppf(Za),4)

# Aplicamos la formula IC
IC1 = round(p-Z*(math.sqrt(p*q/n)),3)
IC2 = round(p+Z*(math.sqrt(p*q/n)),3)

print("P:",p,"| Q:",q,"| a:",a,"| Za:",Za)
print("Z:", Z)
print("Intervalo de confianza:", IC1, "<= pi <=", IC2)

P: 0.26 | Q: 0.74 | a: 0.05 | Za: 0.975
Z: 1.96
Intervalo de confianza: 0.174 <= pi <= 0.346


Con un nivel de confianza del 95% podemos concluir que la verdadera prorcion de bolsas de cafe con menor peso se encuentra en el intervalo de (0.174,0.346)

---

**Pregunta N°4**

Un estudio demostró que 64 de 180 personas que vieron una fotocopiadora anunciada durante la transmisión de un juego de béisbol y 75 de otras 190 personas que la vieron anunciada en un programa de variedades recordaron la marca 2 horas después. Halle un intervalo de confianza al 90% e indique si el programa de variedades tuvo mayor porcentaje de efectividad o no. 

Parametros resaltantes:
- Muestra 1 ($n_1$) = 180
- Numero de exitos 1 ($X_1$) = 64
- Muestra 2 ($n_2$) = 190
- Numero de extios 2 ($X_2$) = 75
- Confianza = 90%

**Formulas a utilizar:**
$$
IC= {p_1-p_2-Z_{(1-\frac{\alpha}{2})}\sqrt{\frac{p_1q_1}{n_1}+\frac{p_2q_2}{n_2}}<\pi_1-\pi_2<p_1-p_2-Z_{(1-\frac{\alpha}{2})}\sqrt{\frac{p_1q_1}{n_1}+\frac{p_2q_2}{n_2}}}
$$

A favor:
$$
p=\frac{X}{n}
$$
En contra:
$$
q = 1 - p 
$$

In [15]:
import scipy.stats as st
import math

# Asignando valores:
n1=180
n2=190
x1=64
x2=75
CF=0.9

p1=x1/n1
p2=x2/n2
q1=1-p1
q2=1-p2
a = round(1 - CF,3)
Za=1-(a/2)

# Calculamos Z
Z = round(st.norm.ppf(Za),4)

# Aplicamos la formula IC
IC1 = round((p1-p2)-Z*(math.sqrt(p1*q1/n1+p2*q2/n2)),3)
IC2 = round((p1-p2)+Z*(math.sqrt(p1*q1/n1+p2*q2/n2)),3)

print("p1:",p1,"p2:",p2,"q1:",q1,"q2:",q2,"a:",a)
print("Za:",Za,"Z:", Z)
print("Intervalo de confianza:", IC1, "<= pi <=", IC2)

p1: 0.35555555555555557 p2: 0.39473684210526316 q1: 0.6444444444444444 q2: 0.6052631578947368 a: 0.1
Za: 0.95 Z: 1.6449
Intervalo de confianza: -0.122 <= pi <= 0.044


Con un 90% de confianza la verdadera diferencia de proporciones está entre -0.122 y 0,044.

---

**Problema N°5**

En el área de producción de una gran embotelladora se desea estudiar la variabilidad del diámetro de la tapa de frasco, para ello se realiza una muestra de 20 envases donde la desviación es de 1.9 mm. Si al 98% de confianza, la desviación es menor a 1.25 mm, no se procede a hacer una revisión de las maquinas, en caso contrario se hará una revisión general. Con la información obtenida ¿que procede?

Parametros resaltantes:
- Muestra 1 ($n$) = 20
- Confianza = 98%

**Formula a utilizar:**
Intervalo de confianza para la desviacion estandar:
$$
\sqrt{\frac{(n-1)S^2}{X^2_{(1-\frac{\alpha}{2},n-1)}}}<\sigma<\sqrt{\frac{(n-1)S^2}{X^2_{(\frac{\alpha}{2},n-1)}}}
$$

In [2]:
import scipy.stats as st
import numpy as np
import math

# Asignando valores:
n=20
CF=0.98
S= 1.9
a= 1 - CF

# Calculamos X usando la tabla Chi-Cuadrado
X1 = round(st.chi2.ppf(1-(a/2),n-1),4)
X2 = round(st.chi2.ppf(a/2,n-1),4)

# Aplicamos la formula IC
IC1 = round(math.sqrt((n-1)*S/X1),3)
IC2 = round(math.sqrt((n-1)*S/X2),3)

print("X1:", X1)
print("X2:", X2)
print("Intervalo de confianza:", IC1, "< σ <", IC2)

X1: 36.1909
X2: 7.6327
Intervalo de confianza: 0.999 < σ < 2.175
